statefun-flink/statefun-flink-core/src/main/java/org/apache/flink/statefun/flink/core/functions/PendingAsyncOperations.java

```diff
@@ -32,18 +31,33 @@
 
 final class PendingAsyncOperations {
 
-  /**
-   * holds the currently in flight and not yet checkpointed async operations.
-   *
-   * <p>A key can be removed from this map in two ways:
-   *
-   * <ul>
-   *   <li>If a remove method was called explicitly (as a result of an async operation completion)
-   *   <li>A checkpoint happens while the async operation is in flight (flush is called). In that
-   *       case the key would be removed from the memoryStore, and written to the backingStore.
-   * </ul>
-   */
-  private final Map<Address, Map<Long, Message>> memoryStore = new HashMap<>();
+  private static final class Key {
```

Comment:
This change makes sense to me. Some day Java will have value types and these classes will be so much cleaner...